src/Illuminate/Validation/Rule.php

```diff
@@ -34,22 +35,30 @@ public static function exists($table, $column = 'NULL')
     /**
      * Get an in constraint builder instance.
      *
-     * @param  array|string  $values
+     * @param  array|string|\Illuminate\Support\Collection  $values
      * @return \Illuminate\Validation\Rules\In
      */
     public static function in($values)
     {
+        if ($values instanceof Collection) {
```

Comment:
Maybe this should be generalized to `Illuminate\Contracts\Support\Arrayable` that `Collection` implements?

Reply:
We'd need to rewrite `In` and `NotIn` rules to accept array and collections then.

Reply:
Since `Arrayable` contains the `toArray` method all you would need to change was the condition I believe.

Reply:
Oh I see you point now. Sorry guys. 

I'd argue that `Arrayable` is implemented by following classes: `Request`, `LengthAwarePaginator`, `MessageBag`, `Fluent` etc.

Still think that `Collection & array` are the way to go.